aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -105,6 +106,14 @@ public String token() {
     return properties().get(OAuth2Properties.TOKEN);
   }
 
+  @Value.Lazy
+  boolean refreshAuthByDefault() {
+    return PropertyUtil.propertyAsBoolean(
+        properties(),
+        CatalogProperties.AUTH_DEFAULT_REFRESH_ENABLED,
+        CatalogProperties.AUTH_DEFAULT_REFRESH_ENABLED_DEFAULT);
```

Comment:
I don't think this is the right property to use. The intent of this property was to refresh a token provided as `token` in config if there was no other information about the token passed in. That's why it is "default refresh enabled" and not just "refresh enabled". When using this for signer, we probably also want to use a signer-specific config property.

Reply:
I would like to encourage reviewers here to focus on what we can do to make things easier to understand & use for users. I understand that this was the original intent of the property, but the code in this area changed so that the current behavior of this property is causing more confusion than necessary. 
I don't think adding a new property helps (the more configuration choices you have, the more difficult it gets). Users that will end up using this property will not understand the difference between this property and a new property. 
Not to blame anyone, but experience shows that most users (including myself) will read documentation as a last resort. They will use the first property they find that implies from its name it does X. Adding new properties here is just making things more difficult.

As I tried to outline in the PR description I believe it is better to use `AUTH_DEFAULT_REFRESH_ENABLED` to fully control the token refresh behavior to not cause any confusion for users. This however implies that we change its default from `false` to `true`.

It would be great to get thoughts and opinions from other folks here as well: @amogh-jahagirdar @jackye1995 @bryanck @danielcweeks

Reply:
If that property is confusing, then let's deprecate and remove it. I agree that its behavior is surprising, but that doesn't mean we should increase confusion by reusing it in other places. The simplest solution to make it more clear is to remove it and rely just on the token expiration property. Then we can add a property for what you actually want to control in this PR.

Reply:
I'm ok deprecating and removing `AUTH_DEFAULT_REFRESH_ENABLED`, but I don't think we increase confusion by using it to control token refresh behavior (because the name itself implied that this is what the property controls). Anyways, I think it makes sense then to introduce `OAuth2Properties.TOKEN_REFRESH_ENABLED` to control the token refresh behavior